homeassistant/components/homekit/util.py

```diff
@@ -110,6 +112,51 @@ def validate_media_player_features(state, feature_list):
     return True
 
 
+SpeedRange = namedtuple('SpeedRange', ('start', 'target'))
+SpeedRange.__doc__ += """: Maps Home Assistant speed \
+values to percentage based HomeKit speeds.
+start: Start of the range (inclusive).
+target: Percentage to use to determine HomeKit percentages \
+from HomeAssistant speed.
+"""
+
+
+class HomeKitSpeedMapping:
+    """Supports conversion between Home Assistant and HomeKit fan speeds."""
+
+    def __init__(self, speed_list):
+        """Initialize a new SpeedMapping object."""
+        if not speed_list or speed_list[0] != fan.SPEED_OFF:
```

Comment:
The case `speed_list is None` shouln't happen. I was thinking if we really need it here if we do it should probably be checked in `type_fans` first, so we don't even add the char. However since it really shouldnkt happen, it's probably save to delete.